core/src/main/java/org/apache/accumulo/core/clientImpl/ActiveScanImpl.java

```diff
@@ -55,6 +55,7 @@ public class ActiveScanImpl extends ActiveScan {
   private Map<String,Map<String,String>> ssio;
   private String user;
   private Authorizations authorizations;
+  private String correlationId;
```

Comment:
Honestly, I think `scanId` is probably much more straight-forward than `correlationId`. Or even "scanLabel".

Reply:
`scanLabel` is attractive.  `scanId` seems like it should be a UUID (like a FATE ID) that uniquely identifies the scan in the system for thing like killing it.  This could be an UUID or maybe just a string that provided the caller / process

Reply:
After reading [coorelationId](https://www.rapid7.com/blog/post/2016/12/23/the-value-of-correlation-ids/) I am more comfortable with correlationId because we can easily signal usage intent and reference other sources if additional clarification is needed.

Reply:
The problem with correlationId, is that it doesn't clue you in to the context of what it is correlating. I can imagine half a dozen different correlationIds in Accumulo... for scans, compactions, RPC requests in general, writes, bulk ingest, etc. They can't all be called correlationId internally. They need to be scoped somehow to the context that says more about *what* they are correlating. That context can be given by the interface the method is attached to, the class the field is in, or additional information in the log message, or some other way. But I think it's important to think about and provide that context somehow, if it isn't present, because "correlationId" by itself is just too generic of a concept that applies to too many contexts.

Reply:
If we migrate to using a specific logger like suggested, then the name of the logger could signal the context.

In a general sense, the correlationId as a general concept looks to want to apply to a more general usage than just a scan. Say that a client reads, possibly updates and then writes the modified or new data within the scope of one unit of work for the client.  When logged consistently, the correlationId could be used to tie all of the log statements for that client operation, scan writes,... across servers and processes.

correlationId seems  synonymous with transactionId in this context, without the implied atomic and rollback properties that are normally associated with transactions.

Reply:
Right, in that sense it's not a scan correlationId, it's a client correlationId.

Reply:
> Right, in that sense it's not a scan correlationId, it's a client correlationId.

So I guess now I'm confused... which one are you going for here? I was under the impression you were going for a scan correlationId, but now I'm wondering if you are intending a client correlationId. If the latter, I would imagine we should focus on amending AccumuloClient.

Reply:
> Say that a client reads, possibly updates and then writes the modified or new data within the scope of one unit of work for the client. When logged consistently, the correlationId could be used to tie all of the log statements for that client operation, scan writes,... across servers and processes.
> Right, in that sense it's not a scan correlationId, it's a client correlationId.

I was just saying that the same correlationId would be used by the client for those operations. I wasn't proposing a correlationId at the client.

Reply:
> > Say that a client reads, possibly updates and then writes the modified or new data within the scope of one unit of work for the client. When logged consistently, the correlationId could be used to tie all of the log statements for that client operation, scan writes,... across servers and processes.
> > Right, in that sense it's not a scan correlationId, it's a client correlationId.
> 
> I was just saying that the same correlationId would be used by the client for those operations. I wasn't proposing a correlationId at the client.

I think I was proposing that, if that's the desired use case we're trying to support.

Reply:
I think a client correlationId should be the goal, this is a step that would not be incompatible with that.

Reply:
> I think a client correlationId should be the goal, this is a step that would not be incompatible with that.

I think adding the APIs to ScannerBase would need to be rolled back if we're going to make it client-wide. It doesn't make sense to add it there if we're going to change it to be added to AccumuloClient later.